### ISSUE TITLE
Segfaults with perl 5.18 and 5.20

### ISSUE THREAD

**Issue by Tekki** [perl5-utils/Hash-Merge#10]
I'm copying the following issue created by @eserte on [rt.cpan.org](https://rt.cpan.org/Public/Bug/Display.html?id=134129) to here because I stumbled over the same problems:

On some of my smoker systems the *-oo.t tests fails with segfaults:
```
...
t/Auto/02-oo.t ......... 
All 87 subtests passed 
...
t/Clone/02-oo.t ........ 
All 87 subtests passed 
...
t/Storable/02-oo.t ..... 
All 87 subtests passed 
...
Test Summary Report
-------------------
t/Auto/02-oo.t       (Wstat: 11 Tests: 87 Failed: 0)
  Non-zero wait status: 11
t/Clone/02-oo.t      (Wstat: 11 Tests: 87 Failed: 0)
  Non-zero wait status: 11
t/Storable/02-oo.t   (Wstat: 11 Tests: 87 Failed: 0)
  Non-zero wait status: 11
```

This seems to happen with perl 5.18 and 5.20. See also http://matrix.cpantesters.org/?dist=Hash-Merge+0.302 --- there is more "red" with these two perl versions.

**Comment by rehsack** [comment]
Yeah, duplicating and quick shots are always strongly encouraged :)

Hash::Merge is a Pure-Perl module, so the core dump comes from somewhere else.

**Comment by Tekki** [comment]
I've copied it here because rt.cpan.org is going to close soon and because on my side the build.log looks exactly the same as the one from Slaven and he was the first to report it.

**Comment by rehsack** [comment]
But he reports any failure he gets without looking into it - regardless of responsibility.

**Comment by Tekki** [comment]
Ok, sorry for the confusion. I've looked a bit into it and the problems come from Storable 2.49_01 that is included in Perl 5.20.3. If I update it to 3.15 first the installation of Hash::Merge succeeds.

**Comment by rehsack** [comment]
That helps, thank you.
